### PR TITLE
Adding comparative operators to reveal widget

### PR DESCRIPTION
**Pull request by ibnishak**
>, <, >=, <=

### CODE REVIEW

File: core/modules/widgets/reveal.js [hunk 1]
```diff
@@ -127,13 +127,34 @@ RevealWidget.prototype.readState = function() {
...
+			break;
 	}
 };
 
 RevealWidget.prototype.readMatchState = function(state) {
 	this.isOpen = state === this.text;
 };
-
+RevealWidget.prototype.readMatchStateltgt = function(state) {
+	this.isOpen = (parseInt(state, 10) < parseInt(this.text, 10))?true:false;
```
Comment (pmario) [inline core/modules/widgets/reveal.js]: what happens if `state` is "open" and not a number?

Comment (pmario) [reply core/modules/widgets/reveal.js]: >this.isOpen = (parseInt(state, 10) < parseInt(this.text, 10))?true:false;

formatting

`this.isOpen = (parseInt(state, 10) < parseInt(this.text, 10)) ? true : false;`

Comment (ibnishak) [reply core/modules/widgets/reveal.js]: NaN returns false

**Comment by ibnishak** [comment]
Added formatting changes suggested by @pmario and documentation for the new **types**

File: core/modules/widgets/reveal.js (hunk 1 above)
Comment (pmario) [reply core/modules/widgets/reveal.js]:
```
parseInt("a", 10) < parseInt("b", 10)
```

Returns false and is logically wrong for users. ... So if we implement something like this, imo we have to have consistent behaviour. 

match "a" == "a"  works
lt "a" < "b" doesn't 

If we restrict the function to numbers, will make the code much simpler, imo we need to implement `eq` too. 
same as `match` but with numbers, ... or ...

we need to implement all possible "alpha numeric" sorting mechanisms, which is a bit of a nightmare.

Comment (ibnishak) [reply core/modules/widgets/reveal.js]: Solved.

Brought AlphaNum to the scope of comparison with the latest commit. 

Now 'a' < 'b'
Also 'a' < 'A'
and 'a 1' <  'a 2'

[Demo](https://ibnishak.github.io/Tesseract/pluginsandmacros/index.html#Reveal-Extended%3A%20Adding%20%3C%2C%3E%2C%3C%3D%2C%3E%3D%20types%20to%20reveal%20widget) with relevant cases

**Comment by Jermolene** [comment]
Thanks @ibnishak

**Comment by ibnishak** [comment]
@Jermolene :+1:

**Pull request merged**